distributed/tensor_parallelism/sequence_parallel_example.py

```diff
@@ -33,51 +39,75 @@
 """
 
 
-def demo_sp(rank, args):
+def demo_sp(args):
     """
     Main body of the demo of a basic version of sequence parallel by using
     PyTorch native APIs.
     """
-    print(f"Running SP example on rank {rank}.")
-    setup(rank, args.world_size)
+    torchrun_setup()
+
+    # understand world topology
+    _rank = int(os.environ["RANK"])
+    _local_rank = int(os.environ["LOCAL_RANK"])
+    _world_size = int(os.environ["WORLD_SIZE"])
+    _local_world_size = int(os.environ["LOCAL_WORLD_SIZE"])
+
+    torch.cuda.set_device(_local_rank)
+
+    def rank_print(msg):
+        """helper function to print only on global rank 0"""
+        if _rank==0:
+            print(f"{msg}")
+
+    print(f"Running basic Megatron style Sequence Parallel example on rank {_rank}.")
```

Comment:
will reduce most of these, but the _rank and rank_print function is to make it cleaner to output the training info etc. to the user rather than generic show every print * every gpu and overwhelm, so do want to keep that.

Reply:
after `init_device_mesh`, you can get a device mesh object and call `device_mesh.get_rank()` to obtain the rank information, I would recommend not using init_process_group call anymore

Reply:
oh I did not realize we can now skip the init_process_group. 
That will definitely help clean up things here.
Will update to that.